### DISCUSSION TITLE
Mention ODIN_ROOT in getting started documention

### DISCUSSION THREAD

**Discussion by MaxGilmore** [odin-lang/Odin#2008] [Ideas/Requests]
Pretty simple but i think ODIN_ROOT enviroment variable should be mentioned in the getting started documentation.

I needed it because i like to handle my path by having one folder with lots of windows shortcut's in it and executing odin.exe through one of these meant ODIN_ROOT was not set correctly. Did kind of get in the way of the language being "easy to start" for me having to mess around with paths a bunch.

**Comment by MaxGilmore** [comment]
Here's my awful windows shortcut setup if anyone ever find's while figuring out their own problems. This doesn't work in PowerShell

Target: %windir%\System32\cmd.exe /c "SET ODIN_ROOT=D:\Tools\Compilers\Odin && ^"D:\Tools\Compilers\Odin\odin.exe^""
Start in: %cd%

**Reply by Kelimion** [reply]
What is the `ODIN_ROOT`  constant set to if you use your shortcut-in-a-folder without using the environment variable, if not the compiler's location?

**Reply by MaxGilmore** [reply]
Ok, I actually investigated it a bit more and this all seems to have nothing to do with ODIN_ROOT.
Made 3 test .lnk's that point directly, go through cmd and goes through cmd with environment variable.
Shortcut via cmd = All are fine
Shortcut via powershell = I think it runs with bad paths in its own window that pops up and you get no output from.
Shortcut via cmd that was launched by powershell = This is where the issue with ODIN_ROOT occurs

Also looks like this isn't the only command line program with this. Seems to be generic windows fun I had somehow never noticed before. I think its probably because I have been using PowerShell more recently and didn't realise dropping into CMD from PowerShell changed behaviour.

> PS D:\Code\OdinHelloWorld> cmd
> Microsoft Windows [Version 10.0.19044.1889]
> (c) Microsoft Corporation. All rights reserved.
> 
> D:\Code\OdinHelloWorld>odin run .\source
> Hellope!
> 
> D:\Code\OdinHelloWorld>odinNoRoot run .\source
> Syntax Error: Invalid path: D:/Code/OdinHelloWorld/core/runtime
> D:/Code/OdinHelloWorld/source/main.odin(3:1) Syntax Error: Invalid path: core:fmt
> 
> D:\Code\OdinHelloWorld>odinDirect run .\source
> Syntax Error: Invalid path: D:/Code/OdinHelloWorld/core/runtime
> D:/Code/OdinHelloWorld/source/main.odin(3:1) Syntax Error: Invalid path: core:fmt
> 
> D:\Code\OdinHelloWorld>PrintShortcut odin
> 
> Shortcut:        D:\Tools\odin.lnk
> Target path:     C:\Windows\System32\cmd.exe
> Arguments:       /c "SET ODIN_ROOT=D:\Tools\Compilers\Odin && ^"D:\Tools\Compilers\Odin\odin.exe^""
> WorkingDirectory:%cd%
> 
> D:\Code\OdinHelloWorld>PrintShortcut odinNoRoot
> 
> Shortcut:        D:\Tools\odinNoRoot.lnk
> Target path:     C:\Windows\System32\cmd.exe
> Arguments:       /c "D:\Tools\Compilers\Odin\odin.exe"
> WorkingDirectory:%cd%
> 
> D:\Code\OdinHelloWorld>PrintShortcut odinDirect
> 
> Shortcut:        D:\Tools\odinDirect.lnk
> Target path:     D:\Tools\Compilers\Odin\odin.exe
> Arguments:
> WorkingDirectory:%cd%

**Reply by Kelimion** [reply]
Looks like maybe PS does something it shouldn't with `argv[0]`, which should point to the executable, not the current working directory?

**Reply by Kelimion** [reply]
Well, it looks like I can run Odin from another location and have it detect its own installation just fine even under PowerShell just fine.
Directory layout:
```
Odin
     odin.exe
Odin-other
     Odin-test
          test.odin
```
```
PS W:\Odin-other> odin run Odin-test
Odin:  W:\Odin\
```
